### PR TITLE
feat: avatar: add avatar popup and fix group tooltips

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:

- Adds `Popup` via `popupProps` to `Avatar` to enable profile card scenarios
- Fixes `Tooltip` in `AvatarGroup` by updating `Avatar` children
- Adds `key` prop to `Avatar`
- Adds `onClick`, `onKeyDown`, `onMouseEnter` and `onMouseLeave` callbacks to `Avatar`
- Improves `Tooltip`, `Popup` keyboard navigation
- Updates `Tooltip` to better separate `Popup` from `Default` variant
- Adds `closeOnReferenceClick` prop to `Tooltip`
- Adds `minHeight` prop to `Tooltip` for better responsive UI support
- Adds `animateOnHover` prop to `AvatarGroup`
- Adds keyboard unit tests to `Popup`
- Normalizes `Tooltip` `disabled` styles
- Improves `useAccessibility` hook to listen for escape key
- Removes `Tooltip` `arrow` padding when `visibleArrow` is `false`
- Fixes `Tooltip` `triggerAbove` to work when `portal`
- Adds `tabIndex` prop to `Popup` and `List`
- Updates `Avatar` and `AvatarGroup` stories
- Fixes `Popup` story layout bug to enable click on `Popup` `children`
- Fixes `maxcount` of basic `AvatarGroup`

https://user-images.githubusercontent.com/99700808/228103689-6f35400b-e24e-4bde-88f5-2180393b60ec.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-40530
ENG-34525

## CHANGE TYPE:

- [X] Bugfix Pull Request
- [X] Feature Pull Request

## TEST COVERAGE:

- [X] Tests for this change already exist
- [X] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Avatar`, `AvatarGroup`, `Popup`, and `Tooltip` stories behave as expected.

### CODE REVIEW

File: src/components/Popup/Popup.stories.tsx [hunk 1]
```diff
@@ -103,9 +103,14 @@ Popups.args = {
...
-          style={{ height: 172, margin: 4, width: 'calc(100% - 8px)' }}
+          style={{
+            height: 172,
+            margin: 4,
+            width: 'calc(100% - 8px)',
+            zIndex: 0,
+          }}
         />
-        <div className="octuple">
+        <div className="octuple" style={{ zIndex: 1 }}>
```
Comment (jhoward-eightfold) [inline src/components/Popup/Popup.stories.tsx:113]: why was this necessary? possible code smell?

File: src/components/Tooltip/tooltip.module.scss [hunk 2]
```diff
@@ -4,16 +4,24 @@ $tooltip-arrow-shadow: 0px 1px 2px rgba(15, 20, 31, 0.12);
   cursor: pointer;
   display: inline-block;
 
-  &.trigger-above {
+  // Ensure portaled tooltip triggers are above when triggerAbove.
+  &.trigger-above[aria-expanded='true'] {
     z-index: $z-index-500;
+
+    &.tooltip {
+      z-index: $z-index-400;
```
Comment (jhoward-eightfold) [inline src/components/Tooltip/tooltip.module.scss:12]: comment for future consideration... i find it easier to reason about zindex values if they are named according to what they representing. then when you have the definitions lined up against each other you can decided if they make sense

something like:
$z-index-tooltip: 400;
$z-index-modal-backdrop: 499;
$z-index-modal: 500;
etc

**Review by jhoward-eightfold — APPROVED** [review]

File: src/components/List/List.tsx [hunk 3]
```diff
@@ -60,6 +61,7 @@ export const List = <T extends any>({
       key={getItemKey(item, index)}
       className={itemClasses}
       style={itemStyle}
+      tabIndex={tabIndex}
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/List/List.tsx:64]: Tested keyboard and for whatever reason Chrome included <li> in the tab indices, this provides a workaround to some browser quirks.

File: src/components/Popup/Popup.stories.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Popup/Popup.stories.tsx:113]: Yes, need to investigate the `Stack` component behavior. May be isolated to Surface UI wrappers only.  Tracking with ENG-48127.
